src/quest/manager.py

```diff
@@ -86,28 +94,47 @@ def _remove_workflow(self, workflow_id: str):
         self._workflow_data.remove(data)
 
     def _start_workflow(self,
-                        workflow_type: str, workflow_id: str, workflow_args, workflow_kwargs,
-                        background=False):
+                        workflow_type: str, workflow_id: str, workflow_args, workflow_kwargs, background=False):
         workflow_function = self._create_workflow(workflow_type)
 
+        future = asyncio.Future()
+        self._futures[workflow_id] = future
```

Comment:
If you don't use `future` elsewhere in this function, make this a direct assignment:
```py
self._futures[workflow_id] = asyncio.Future()
```